pkg/handlers/defaulthandler.go

```diff
@@ -0,0 +1,245 @@
+package handlers
```

Comment:
Long-term I think this should be split into `DefaultHandler` and `ArchiveHandler`. I know the name change was made to minimize changes, but leaving it as `archive.go` for this PR would make it a lot easier to see and comment what was changed. Also name should be "Default" since it's part of the handlers package already.

Reply:
I have a follow up PR to address this issue.